### PR TITLE
Add Tag Template module

### PR DESCRIPTION
**Pull request by lcaggio**
Create Data Catalog - Tag Template module. The module will handle the creation of tag templates and the corresponding factory.

I applicable, I acknowledge that I have:
- [X] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [X] Ran `terraform fmt` on all modified files
- [X] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [X] Made sure all relevant tests pass

### CODE REVIEW

File: modules/data-catalog-tag-template/main.tf [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * limitations under the License.
+ */
+
+locals {
+  _factory_tag_template = {
+    for f in fileset(var.factory_config_path, "**/*.yaml") :
+    trimsuffix(f, ".yaml") => yamldecode(file("${var.factory_config_path}/${f}"))
+  }
+
+  _tag_template_default = {
```
Comment (lcaggio) [inline modules/data-catalog-tag-template/main.tf:23]: is there a better way to handle defaults?

Comment (wiktorn) [reply modules/data-catalog-tag-template/main.tf:23]: I prefer:
```hcl
  is_required = try(fields.value["is_required"], false)
```
To:
```hcl
  is_required = try(fields.value["is_required"], local._tag_template_default.fields.is_required)
```

As you can see the default value exactly where it is used.

Comment (lcaggio) [reply modules/data-catalog-tag-template/main.tf]: I see, thanks. Updated.

File: modules/data-catalog-tag-template/variables.tf [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "factory_config_path" {
+  description = "Path to data files and folders that enable factory functionality."
+  type        = string
```
Comment (ludoo) [inline modules/data-catalog-tag-template/variables.tf:19]: Can you align to this naming and type?

https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/modules/organization/variables.tf#L31

File: modules/data-catalog-tag-template/variables.tf [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+variable "tag_templates" {
+  description = "Tag templates definitions in the form {TAG_TEMPLATE_ID => TEMPLATE_DEFINITION}."
+  type = map(object({
+    display_name = optional(string)
+    force_delete = optional(bool, false)
+    region       = string
+    fields = list(object({
```
Comment (ludoo) [inline modules/data-catalog-tag-template/variables.tf:70]: can fields be changed/modified after creation? if yes, this should be a map keyed maybe on field_id

File: modules/data-catalog-tag-template/variables.tf (hunk 2 above)
Comment (lcaggio) [reply modules/data-catalog-tag-template/variables.tf]: done

File: modules/data-catalog-tag-template/variables.tf (hunk 3 above)
Comment (lcaggio) [reply modules/data-catalog-tag-template/variables.tf]: updated

**Review by ludoo — APPROVED** [review]

**Pull request merged**